tus_storagehandler/app.py

```diff
@@ -0,0 +1,72 @@
+"""API server entry point."""
+
+import logging
+import os
+from pathlib import Path
+from typing import Optional
+
+from connexion import FlaskApp
+from foca import Foca
+
+logger = logging.getLogger(__name__)
+
+
+def init_app() -> FlaskApp:
+    """Initialize and return the FOCA app.
+
+    This function initializes the FOCA app by loading the configuration
+    from the environment variable `TUS_FOCA_CONFIG_PATH` if set, or from
+    the default path if not. It raises a `FileNotFoundError` if the
+    configuration file is not found.
+
+    Returns:
+        A Connexion application instance.
+
+    Raises:
+        FileNotFoundError: If the configuration file is not found.
+    """
+    # Determine the configuration path
+    if config_path_env := os.getenv("TUS_FOCA_CONFIG_PATH"):
+        print(config_path_env)
+        config_path = Path(config_path_env).resolve()
+    else:
+        config_path = (
+            Path(__file__).parents[1] / "deployment" / "config.yaml"
+        ).resolve()
+
+    # Check if the configuration file exists
+    if not config_path.exists():
+        raise FileNotFoundError(f"Config file not found at: {config_path}")
+
+    foca = Foca(
+        config_file=config_path,
+    )
+    return foca.create_app()
+
+
+def main(env: Optional[str] = None) -> None:
+    """Run FOCA application.
+
+    Args:
+        env (str, optional): The environment in which to run the application.
+                             Defaults to 'prod' if not specified.
+    """
+    # Set default port
+    default_port = 8081
+
+    # Initialize application
+    app = init_app()
+
+    # Set environment and port
+    if env is None:
+        env = "prod"
+    port = app.port if env == "prod" else default_port
+
+    # Run the application
+    app.run(port=port)
+
+
+if __name__ == "__main__":
+    # Get environment variable or default to 'dev'
+    environment = os.getenv("ENV", "dev")
```

Comment:
Document env variable as well.

Reply:
Not the best name, perhaps, given that `env` lists all environment variables in the shell, so `$ENV` is a bit ambiguous. Better to use `TUS_ENV` (now) or `CSH_ENV` (later, see other comments).

Reply:
Since this env is related to foca running on diff ports maybe use a general name which can be adopted by other repos too. Something like 'ENVIRONMENT'?

Reply:
I don't agree: If I have more than one app deployed on the same machine, with a project-independent env name, I will not be able to set different values for different projects. Or am I missing something - @JaeAeich?

Reply:
Actually, after looking at the code, I don't really see the point of this variable at all. The port can be configured via `server.port` in `config.yaml` (currently `8080`). FOCA will then set `app.port` to the specified value. And if that's not enough, there are also `server.debug` (available via `app.debug` and `app.app.config["DEBUG"]` or `app.app.config.foca.server.debug`) and `server.environment` (available via `app.app.config["ENVIRONMENT"]` or `app.app.config.foca.server.environment`).

Sure it would be nice to have env vars for all the FOCA params. But that is something that we should set up in FOCA at some point, IMO. Besides, overriding the port for production environments in a way that can not be configured (hardcoded to `8081`) is surely not the way to go.

So I think it's best to remove the variable entirely and let FOCA take care of setting the port. If you want a different port for development, keep your private copy of the app/FOCA config (maybe outside of the repo or add it to your personal Git ignore so that you don't accidentally commit it), change the port to whatever you like and pass it via `CSH_FOCA_CONFIG_PATH`.